daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -344,11 +370,23 @@ class VariableMap private(vTable: Map[GlobalQName, ArrayBuffer[VariableInstance]
    */
   def readVariable(vrd: VariableRuntimeData, referringContext: ThrowsSDE, maybeState: Maybe[ParseOrUnparseState]): DataValuePrimitive = {
     val varQName = vrd.globalQName
+    if (maybeState.isDefined) {
```

Comment:
So this actually seems like it might be an artifact of dstate.parseOrUnparseState returning a maybe. For the time being I've modified the call Dstate makes when reading a variable to do an Assert.invariant(dstate.parseOrUnparserState.isDefined)

Reply:
After spending too much time trying to figure out how to make a fake PState (couldn't find any existing code and PState seems to require a lot of scaffolding to get set up) I decided to modify the test slightly to simply call vmap.find() and check the value of the returned variable instance, which doesn't require a PState.